pkg/component/runtime/service_command.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/dolmen-go/contextio"
+
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+	"github.com/elastic/elastic-agent/pkg/core/process"
+)
+
+func executeCommand(ctx context.Context, log *logger.Logger, binaryPath string, args []string, env []string, timeout time.Duration) error {
+	log = log.With("context", "command output")
+	// Create context with timeout if the timeout is greater than 0
+	if timeout > 0 {
+		var cn context.CancelFunc
+		ctx, cn = context.WithTimeout(ctx, timeout)
+		defer cn()
+	}
+
+	opts := []process.StartOption{
+		process.WithContext(ctx),
+		process.WithArgs(args),
+		process.WithEnv(env),
+	}
+
+	// Set the command working directory from binary
+	// This is needed because the endpoint installer was looking for it's resources in the current working directory
+	wdir := filepath.Dir(binaryPath)
+	if wdir != "." {
```

Comment:
this should never happen (wdir to be `.`) we cannot use relative paths, but i dont mind having it here.